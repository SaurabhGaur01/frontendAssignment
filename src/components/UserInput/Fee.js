import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setFee } from '../../ducks/fee';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const Fee = ({
    actionSetFee,
    fee, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputFee = parseInt(event.target.value);
        actionSetFee(inputFee);
    }

    return (
        <div>
            Fee:
            <input id= "fee" type="text" name="fee" defaultValue={fee} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

Fee.propTypes = {
    actionSetFee: PropTypes.func.isRequired,
    fee: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetFee: setFee,
}

export const mapStateToProps = state => ({
    fee: state.fee,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { Fee as TestableFee };

export default hocChain(Fee);
