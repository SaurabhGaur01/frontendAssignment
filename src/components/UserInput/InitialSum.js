import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setInitialSum } from '../../ducks/initialSum';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const InitialSum = ({
    actionSetInitialSum,
    initialSum, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputInitialSum = parseInt(event.target.value);
        actionSetInitialSum(inputInitialSum);
    }

    return (
        <div>
            Initial Sum:
            <input id= "initialSum" type="text" name="initialSum" defaultValue={initialSum} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

InitialSum.propTypes = {
    actionSetInitialSum: PropTypes.func.isRequired,
    initialSum: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetInitialSum: setInitialSum,
}

export const mapStateToProps = state => ({
    initialSum: state.initialSum,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { InitialSum as TestableInitialSum };

export default hocChain(InitialSum);
