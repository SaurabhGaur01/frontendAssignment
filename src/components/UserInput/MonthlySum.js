import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setMonthlySum } from '../../ducks/monthlySum';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const MonthlySum = ({
    actionSetMonthlySum,
    monthlySum, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputMonthlySum = parseInt(event.target.value);
        actionSetMonthlySum(inputMonthlySum);
    }

    return (
        <div>
            Monthly Sum:
            <input type="text" id="monthlySum" name="monthlySum" defaultValue={monthlySum} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

MonthlySum.propTypes = {
    actionSetMonthlySum: PropTypes.func.isRequired,
    monthlySum: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetMonthlySum: setMonthlySum,
}

export const mapStateToProps = state => ({
    monthlySum: state.monthlySum,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { MonthlySum as TestableMonthlySum };

export default hocChain(MonthlySum);
