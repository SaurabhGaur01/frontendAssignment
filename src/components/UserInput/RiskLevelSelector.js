import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setRiskLevel } from '../../ducks/riskLevel';

const maxRiskLevel = 25;
const defaultRisk = 10;

const RiskLevelSelector = ({
    actionSetRiskLevel, 
}) => {
    const onChange = (event) => {
        var inputRiskLevel = parseInt(event.target.value);
        actionSetRiskLevel(inputRiskLevel);
    }

    const options = [];
    for(let k=1; k<=maxRiskLevel; ++k) {
        options.push(
            <option key={k} value={k}>{k}</option>
        );
    }

    return (
        <div>
            Risk level:
            <select id="riskLevel" className="riskLevel" onChange={onChange} defaultValue={defaultRisk}>
                {options}
            </select>
        </div>
    );
}

RiskLevelSelector.propTypes = {
    actionSetRiskLevel: PropTypes.func.isRequired,
};

const mapDispatchAsProps = {
    actionSetRiskLevel: setRiskLevel,
}

const hocChain = compose(
    connect(null, mapDispatchAsProps),
);

export { RiskLevelSelector as TestableRiskLevelSelector };

export default hocChain(RiskLevelSelector);
