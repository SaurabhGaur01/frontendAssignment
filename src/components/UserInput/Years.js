import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setYears } from '../../ducks/years';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const Years = ({
    actionSetYears,
    years, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputYears = parseInt(event.target.value);
        actionSetYears(inputYears);
    }

    return (
        <div>
            Years:
            <input type="text" id="years" name="years" defaultValue={years} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

Years.propTypes = {
    actionSetYears: PropTypes.func.isRequired,
    years: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetYears: setYears,
}

export const mapStateToProps = state => ({
    years: state.years,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { Years as TestableYears };

export default hocChain(Years);
