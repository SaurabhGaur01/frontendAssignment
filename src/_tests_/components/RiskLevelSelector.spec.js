import React from 'react';
import { shallow } from 'enzyme';
import { TestableRiskLevelSelector } from '../../components/UserInput/RiskLevelSelector';

describe('RiskLevelSelector Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            actionSetRiskLevel: jest.fn(),
        };
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableRiskLevelSelector {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    describe('onChange', () => {
        it('should call actionSetRiskLevel when user input any value', () => {
            const wrapper = shallow(<TestableRiskLevelSelector {...props} />);
            const event = { target: { value: 10}};
            wrapper.find('#riskLevel').simulate('change', event);
            expect(props.actionSetRiskLevel).toHaveBeenCalledTimes(1);
            expect(props.actionSetRiskLevel).toHaveBeenCalledWith(10);
        })
    })
})