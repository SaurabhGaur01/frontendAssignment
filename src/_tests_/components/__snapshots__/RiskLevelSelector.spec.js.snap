// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskLevelSelector Component should render correctly 1`] = `
<div>
  Risk level:
  <select
    className="riskLevel"
    defaultValue={10}
    id="riskLevel"
    onChange={[Function]}
  >
    <option
      key="1"
      value={1}
    >
      1
    </option>
    <option
      key="2"
      value={2}
    >
      2
    </option>
    <option
      key="3"
      value={3}
    >
      3
    </option>
    <option
      key="4"
      value={4}
    >
      4
    </option>
    <option
      key="5"
      value={5}
    >
      5
    </option>
    <option
      key="6"
      value={6}
    >
      6
    </option>
    <option
      key="7"
      value={7}
    >
      7
    </option>
    <option
      key="8"
      value={8}
    >
      8
    </option>
    <option
      key="9"
      value={9}
    >
      9
    </option>
    <option
      key="10"
      value={10}
    >
      10
    </option>
    <option
      key="11"
      value={11}
    >
      11
    </option>
    <option
      key="12"
      value={12}
    >
      12
    </option>
    <option
      key="13"
      value={13}
    >
      13
    </option>
    <option
      key="14"
      value={14}
    >
      14
    </option>
    <option
      key="15"
      value={15}
    >
      15
    </option>
    <option
      key="16"
      value={16}
    >
      16
    </option>
    <option
      key="17"
      value={17}
    >
      17
    </option>
    <option
      key="18"
      value={18}
    >
      18
    </option>
    <option
      key="19"
      value={19}
    >
      19
    </option>
    <option
      key="20"
      value={20}
    >
      20
    </option>
    <option
      key="21"
      value={21}
    >
      21
    </option>
    <option
      key="22"
      value={22}
    >
      22
    </option>
    <option
      key="23"
      value={23}
    >
      23
    </option>
    <option
      key="24"
      value={24}
    >
      24
    </option>
    <option
      key="25"
      value={25}
    >
      25
    </option>
  </select>
</div>
`;
